src/UI/Art/SpriteArtist/CommanderOverlayArtist.java

```diff
@@ -7,108 +7,122 @@
-      g.drawImage(overlayImage, xPos, verticalOffset, overlayImage.getWidth(), overlayImage.getHeight(), null);
-      drawIconAndValue(g, SpriteLibrary.MapIcons.ENERGY, energy, valueLength, valueXPos, yTextOffset);
-      drawIconAndValue(g, SpriteLibrary.MapIcons.FUNDS,  money,  valueLength, valueXPos, SpriteLibrary.baseIconSize + yTextOffset);
-      int pbXPos = mapViewWidth - powerBarImage.getWidth();
-      if( trueIfBarFalseIfText )
+      // Choose left or right overlay image to draw.
+      BufferedImage overlayImage = SpriteLibrary.getCoOverlay(commander, overlayIsLeft);
+      BufferedImage powerBarImage = null;
+      CommanderAbility ability = commander.getActiveAbility();
+      final int xAbilityOffset;
+      if( null == ability )
       {
-        // We are drawing the power bar, and want to flip it horizontally.
-        g.drawImage( powerBarImage, pbXPos + (powerBarImage.getWidth()), (overlayImage.getHeight()+verticalOffset) - (POWERBAR_BUFFER), -powerBarImage.getWidth(), powerBarImage.getHeight(), null );
+        powerBarImage = buildCoPowerBar(commander, commander.getAbilityCosts(), commander.getAbilityPower(), overlayIsLeft);
+        xAbilityOffset = coEyesWidth + 1;
       }
       else
       {
-        // We are drawing an ability name, and we don't want to flip it.
-        g.drawImage( powerBarImage, pbXPos, (overlayImage.getHeight()+verticalOffset) - (POWERBAR_BUFFER), powerBarImage.getWidth(), powerBarImage.getHeight(), null );
+        powerBarImage = getActiveAbilityName(ability.toString());
+        xAbilityOffset = xTextOffset;
       }
+
+      int energy = COStateInfo.getEnergyUntilNextPower(commander);
+      int energyLength = ("" + energy).length();
+
+      final int yPowerBarOffset = SpriteLibrary.baseIconSize + yTextOffset - 1 + yCurrentOffset;
+      if( overlayIsLeft )
+      { // Draw the overlay on the left side.
+        g.drawImage(overlayImage, 0, yCurrentOffset, overlayImage.getWidth(), overlayImage.getHeight(), null);
+        drawIconAndValue(g, SpriteLibrary.MapIcons.ENERGY, energy, energyLength, xTextOffset, yTextOffset + yCurrentOffset);
+        g.drawImage( powerBarImage, xAbilityOffset, yPowerBarOffset, powerBarImage.getWidth(), powerBarImage.getHeight(), null );
+      }
+      else
+      { // Draw the overlay on the right side.
+        int xPos = mapViewWidth - overlayImage.getWidth();
+        int valueXPos = mapViewWidth - ICON_VALUE_SPACING - spriteA.getWidth()*energyLength - xTextOffset;
+        g.drawImage(overlayImage, xPos, yCurrentOffset, overlayImage.getWidth(), overlayImage.getHeight(), null);
+        drawIconAndValue(g, SpriteLibrary.MapIcons.ENERGY, energy, energyLength, valueXPos, yTextOffset + yCurrentOffset);
+        int pbXPos = mapViewWidth - xAbilityOffset - powerBarImage.getWidth();
+        g.drawImage( powerBarImage, pbXPos, yPowerBarOffset, powerBarImage.getWidth(), powerBarImage.getHeight(), null );
+      }
```

Comment:
Gotta say, I think I like the power bar/funds better where they used to be. It looks more natural if the bar extends past the end of the overlay, and looks more natural than the brown funds box that's replaced it.

The only downsides of the old scheme are 1) using slightly more vertical space to put a power bar between each CO overlay and 2) Repeating the funds count across multiple COs if they are sharing funds. I don't think either of those is a huge deal though. Plus, isn't there a planned tag mode without shared funds?

Reply:
> It looks more natural if the bar extends past the end of the overlay, and looks more natural than the brown funds box that's replaced it.

Very solid points.
I am considering moving to a "star"-based ability energy storage scheme, since I think I can support all the use-cases we care about with it. That's not really here or there, though.
> 1) using slightly more vertical space to put a power bar between each CO overlay

There's also the question of whether we should have transparent gaps in there or have the bar pips overlap with the next CO header. ...Thinking about it, I guess it wouldn't be a big deal to just overlap, since they're drawn top-to-bottom (so the top of each header would be an unbroken line).
> Repeating the funds count across multiple COs if they are sharing funds.

This is the main reason I moved things around. I don't like duplicate data == wasted space.
I suppose we could pull the power bars *and* gold out, and put the CO name or something else in.
> Plus, isn't there a planned tag mode without shared funds?

No. If there is, then I'm gonna have to refactor 'cause `Army` owns the purse at present.